### PR TITLE
DOC: ecosystem.rst: Koalas

### LINKED ISSUE #26424: Add Koalas project in pandas Ecosystem
#### Code Sample, a copy-pastable example if possible

N/A

#### Problem description

There is a project called Koalas which targets the distributed Pandas DataFrame on the top of Apache Spark.

It might be good to document this as one of the eco systems too.

#### Expected Output

Koalas description at https://pandas.pydata.org/pandas-docs/stable/ecosystem.html

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Darwin
OS-release: 18.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 4.4.1
pip: 19.0.3
setuptools: 40.8.0
Cython: None
numpy: 1.16.2
scipy: None
pyarrow: 0.13.0
xarray: None
IPython: None
sphinx: 2.0.1
patsy: None
dateutil: 2.8.0
pytz: 2019.1
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

### PR DESCRIPTION
**Pull request by HyukjinKwon**
- [x] closes #26424
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I (almost) copied and pasted from its README.md.

### CODE REVIEW

**Review by datapythonista — CHANGES_REQUESTED** [review]
I see that you took the description from Koalas documentation, but it sounds more like a sales speech than a useful description (e.g. `be more productive when interacting with big data`).

Also, I think we can safely assume that users checking this documentation know what pandas is (and probably spark).

Can you please rephrases, being concise on what Koalas does. Also, didn't check Koalas, but I doubt it implements the full pandas API with spark as a backend. Please add what users can expect in terms of stability and completeness of the API.

**Comment by HyukjinKwon** [comment]
Sure, will do. Thanks!

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -285,6 +285,11 @@ provides a familiar ``DataFrame`` interface for out-of-core, parallel and distri
 
 Dask-ML enables parallel and distributed machine learning using Dask alongside existing machine learning libraries like Scikit-Learn, XGBoost, and TensorFlow.
 
+`Koalas <https://koalas.readthedocs.io/en/latest/>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Koalas provides a familiar pandas DataFrame interface on top of Apache Spark. It enables users to leverage multi-cores on one machine or a cluster of machines to speed up or scale their DataFrame code.
+
```
Comment (HyukjinKwon) [inline doc/source/ecosystem.rst]: @datapythonista, how does it looks like?

Comment (datapythonista) [reply doc/source/ecosystem.rst]: That is much better in my opinion. Let's see if anyone else has feedback, but I think this gives much more relevant information to the readers than before. Thanks!

**Review by datapythonista — APPROVED** [review]
lgtm

**Review by WillAyd — APPROVED** [review]
lgtm as well

**Review by TomAugspurger — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thank you, @datapythonista, @WillAyd and @TomAugspurger for your guidance and approval.

**Pull request merged**